### PR TITLE
Create build, test, publish, and secure GitHub Action workflows for .NET apps

### LINKED ISSUE #24863: Create new GitHub Action articles
Create new GitHub Action articles:

- [ ] Overview - actions, workflows, etc
- [ ] Build - validation workflow
- [ ] Test - consider test results visibility
- [ ] Publish - to Azure of course
- [ ] Secure - CodeQL

Add them to the existing _/dotnet/devops_ directory and pair them with the [Tutorial: Create a GitHub Action with .NET](https://docs.microsoft.com/dotnet/devops/create-dotnet-github-action).

- Do **_not_** duplicate the [GitHub content](https://docs.github.com/actions/guides/building-and-testing-net), but rather make it complementary and link to .NET CLI.
- Do **_not_** duplicate the [e-book content](https://docs.microsoft.com/dotnet/architecture/devops-for-aspnet-developers/), instead make more applicable to all .NET workloads -- not just ASP.NET Core.

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

- Create build, test, publish, and secure GitHub Action workflows for .NET apps.

Fixes #24863

## Internal previews

✔️ [Overview](https://review.docs.microsoft.com/en-us/dotnet/devops/github-actions-overview?branch=pr-en-us-24924)
✔️ [Build](https://review.docs.microsoft.com/en-us/dotnet/devops/dotnet-build-github-action?branch=pr-en-us-24924)
✔️ [Publish](https://review.docs.microsoft.com/en-us/dotnet/devops/dotnet-publish-github-action?branch=pr-en-us-24924)
✔️ [CodeQL](https://review.docs.microsoft.com/en-us/dotnet/devops/dotnet-secure-github-action?branch=pr-en-us-24924)
✔️ [Test](https://review.docs.microsoft.com/en-us/dotnet/devops/dotnet-test-github-action?branch=pr-en-us-24924)

## TOC updates

image

### CODE REVIEW

**Comment by IEvangelist** [comment]
@Youssef1313 - another thing we should consider for the false negatives being reported of redirections:

- If a file has never been published before (not sure how we'd figure that out), but these are all new files, and when I moved them the GitHub Action thought that they're being deleted -- and require a redirection.

**Comment by Youssef1313** [comment]
> but these are all new files

@IEvangelist They are not new, but are indeed false positives. I think we could do fix that by ignoring these patterns:

https://github.com/dotnet/docs/blob/bc8e38479d9867c7ed4b308be596ee7642422754/docfx.json#L17-L23

I'll look into that soon.

**Review by adegeo — COMMENTED** [review]
For repo structure, your snippets/images should be in either a subfolder either named after [the article using it](https://docs.microsoft.com/en-us/contribute/dotnet/dotnet-contribute#example-folder-structure) or in a folder named *shared* (if shared across multiple articles).

You should also provide a VB.NET project/snippets too 😁

**Comment by IEvangelist** [comment]
> For repo structure, your snippets/images should be in either a subfolder either named after [the article using it](https://docs.microsoft.com/en-us/contribute/dotnet/dotnet-contribute#example-folder-structure) or in a folder named _shared_ (if shared across multiple articles).
> 
> You should also provide a VB.NET project/snippets too 😁

The only C# files are ones that have been moved, no VB to add.

And for the folder structure -- my concern is that the folder name "snippets" ends up being redundant, I'd rather have a single folder named "snippets" under this "devops" directory, with each sub folder being contextual.

```
devops/
  - includes/
  - snippets/
    - build/
    - test/
    - publish/
    - codeql 
  - media/
```

As compared to:

```
devops/
  - includes/
  - build/
    - snippets/
  - test/
    - snippets/
  - publish/
    - snippets/
  - codeql
    - snippets/
  - media/
```

I've been scoping includes/snippets/media directories to contextual folders now for a while -- following this pattern. I agree we should be a bit more consistent, but I'd rather update the contributor guidance to suggest an alternative approach or to say use your best judgement. Any thoughts on this?

**Comment by adegeo** [comment]
You're correct in your structure assumption, but misinterpreting what I'm saying 😄 What I mean is the name of those folders should match the article, or be named shared if it's spread across multiple articles:

```
devops/
  - includes/
  - snippets/
    - dotnet-publish-github-action/
    - dotnet-test-github-action/
    - dotnet-secure-github-action/
  - media/
    - shared/
    - dotnet-publish-github-action/
    - dotnet-test-github-action/
    - dotnet-secure-github-action/
```

Regarding VB, we should file an issue to add VB to these articles, since it's supposed to be supported by our docs (except in the csharp area of course)

File: docs/devops/github-actions-overview.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+## GitHub Actions
+
+GitHub Actions represent standalone commands, such as:
+
+- [actions/checkout](https://github.com/actions/checkout) - This action checks-out your repository under `$GITHUB_WORKSPACE`, so your workflow can access it.
+- [actions/setup-dotnet](https://github.com/actions/setup-dotnet) - This action sets up a .NET CLI environment for use in actions.
+- [dotnet/versionsweeper](https://github.com/dotnet/versionsweeper) - This action sweeps .NET repos for out-of-support target versions of .NET.
+
+> [!TIP]
+> The .NET docs team is utilizing the .NET version sweeper GitHub Action to automate issue creation. It runs on a schedule (as a CRON job). When it detects that .NET projects are targeting out-of-support versions &mdash; it creates issues reporting its findings. This is configurable, and helpful for tracking .NET version support concerns. It's available in the [GitHub Action Marketplace](https://github.com/marketplace/actions/net-version-sweeper).
```
Comment (BillWagner) [inline docs/devops/github-actions-overview.md:23]: I don't think "cron" should be capitalized.  I'd check with @tdykstra to be sure.

**Review by BillWagner — APPROVED** [review]
This LGTM @IEvangelist 

I had one question to check on before merging, and then when the existing discussions are complete, you can :shipit:

File: docs/devops/github-actions-overview.md (hunk 1 above)
Comment (tdykstra) [reply docs/devops/github-actions-overview.md:23]: I would use lowercase -- cron.

**Pull request merged**